source/agora/consensus/protocol/Nominator.d

```diff
@@ -1034,36 +1028,46 @@ extern(D):
         this.network.gossipBlockSignature(block_sig);
     }
 
-    /// Were we active in this block (i.e. we are not catching up after being offline)
-    public bool safeToSign (in Height height)
-        nothrow
-    {
-        return !!(height in this.slot_sigs); // true if we were active in nomination during this block
-    }
+    /***************************************************************************
 
-    /// Create a combined Schnorr multisig and return the updated block
-    private Block updateMultiSignature (in Block block) @safe
+        Add missing block signatures to provided block header if known
+
+        Params:
+            header = header to be updated
+
+        Returns:
+            the updated header or `BlockHeader.init`` if we have no signatures
+
+    ***************************************************************************/
+
+    public const(BlockHeader) updateMultiSignature (in BlockHeader header) @safe
```

Comment:
By returning `const(BlockHeader)` here we can return the original header if we return early.